src/kaleidoscope/Device.h

```diff
@@ -65,6 +65,8 @@ class Device {
 
  public:
 
+  typedef _DeviceDescription Description;
```

Comment:
I suggest this typedef which enables writing `Device::Description`. Otherwise it is not possible to access the description if only the device class is available.